src/dependency-manager/src/component-config/base.ts

```diff
@@ -50,7 +53,22 @@ export abstract class ComponentConfig extends BaseSpec {
     return classToClass(this);
   }
 
+  expand() {
```

Comment:
I like this a lot. 

My static instincts would have me write this method like this:

```
static expand(config: ComponentConfig);
```
And call it like this:

```
const expanded_config = ComponentConfig.expand(some_config);
```

This makes it clear that `config.expand()` isn't actually modifying any state on the `config`, which is what I was initially wondering. However, I know we don't generally follow that pattern so consider this a soft recommendation.

Reply:
Let me know if this helps and you like the use of JSDoc annotations. Thinking we should use more for critical code.
https://github.com/architect-team/architect-cli/pull/255/commits/261ba6a936fc89cf2166af45685be46c2fa9e4fd#diff-7cc7e5ad27691772a28e4dd5c45ae372R28

<img width="436" alt="Capture" src="https://user-images.githubusercontent.com/737712/88245456-446d5300-cc65-11ea-8360-6efa053381e4.PNG">

Reply:
More info here: https://code.visualstudio.com/docs/languages/typescript#_jsdoc-support

Reply:
Yes, I do like JSDocs to clarify this. It still doesn't replace it being static (and literally barring it from modifying underlying object state) but it does help clarify intent.